SU-HSDO/suhumsci
# STN-982: Mega Menu On-Click Modification

# READY FOR REVIEW

## Summary
Currently, the mega menu is hover-based, in that dropdown menus are displayed when the user hovers over a menu item. The desired effect is that the menu be click-based instead, so this PR changes the functionality so that dropdown menus are displayed when a menu item is clicked instead.

[See Jira card](https://sparkbox.atlassian.net/browse/STN-982).

## Need Review By (Date)
asap

## Urgency
medium

## Steps to Test
1. `git checkout STN-982--megamenu-onclick`
2. `lando drush @sparkbox_sandbox.local cr `
3. `npm start`
4. Place a `Drupal 8 Mega Menu` block in Structure | Block Layout | "Menu" region if the Mega Menu does not already exist on your local environment.
5. Click on menus items with a dropdown indicator. Verify the following behaviors:
  - Dropdown menus display on click, not on hover
  - Dropdown indicators animate and point upward when the menu is open, animate and point down when either the menu item is clicked again, or the menu item is clicked off of (loses focus). 
  - Dropdown menus close when the menu item is clicked off of (loses focus)
  - An open dropdown menu does not close when another menu item is hovered
  - If another menu item is clicked, the open dropdown menu is closed, and the corresponding menu item dropdown is opened
6. Verify that the menu is keyboard accessible. Tab to navigate through menu items, and Enter to open a dropdown menu
7. Resize the screen to verify that the menu stays open. This does not include mobile, as the mega menu is not shown on mobile screens

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
